indico/migrations/versions/20191023_1820_2496c4adc7e9_add_track_principal.py

```diff
@@ -0,0 +1,304 @@
+        ),
+        sa.ForeignKeyConstraint(
+            ['local_group_id'],
+            [u'users.groups.id'],
+            name=op.f('fk_track_principals_local_group_id_groups'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['track_id'],
+            [u'events.tracks.id'],
+            name=op.f('fk_track_principals_track_id_tracks'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['user_id'],
+            [u'users.users.id'],
+            name=op.f('fk_track_principals_user_id_users'),
+        ),
+        sa.PrimaryKeyConstraint('id', name=op.f('pk_track_principals')),
+        schema='events',
+    )
+
+    op.create_index(
+        op.f('ix_track_principals_mp_group_provider_mp_group_name'),
+        'track_principals',
+        ['mp_group_provider', 'mp_group_name'],
+        unique=False,
+        schema='events',
+    )
+    op.create_index(
+        'ix_uq_track_principals_email',
+        'track_principals',
+        ['email', 'track_id'],
+        unique=True,
+        schema='events',
+        postgresql_where=sa.text(u'type = 4'),
+    )
+    op.create_index(
+        'ix_uq_track_principals_local_group',
+        'track_principals',
+        ['local_group_id', 'track_id'],
+        unique=True,
+        schema='events',
+        postgresql_where=sa.text(u'type = 2'),
+    )
+    op.create_index(
+        'ix_uq_track_principals_mp_group',
+        'track_principals',
+        ['mp_group_provider', 'mp_group_name', 'track_id'],
+        unique=True,
+        schema='events',
+        postgresql_where=sa.text(u'type = 3'),
+    )
+    op.create_index(
+        'ix_uq_track_principals_user',
+        'track_principals',
+        ['user_id', 'track_id'],
+        unique=True,
+        schema='events',
+        postgresql_where=sa.text(u'type = 1'),
+    )
+
+    _upgrade_permissions()
+
+    op.drop_table('track_conveners', schema='events')
+    op.drop_table('track_abstract_reviewers', schema='events')
+
+
+def _create_abstract_reviewers_entry(conn, user_id, track_id=None, event_id=None):
+    conn.execute('''
+        INSERT INTO events.track_abstract_reviewers
+        (user_id, track_id, event_id) VALUES
+        (%s,       %s,      %s)
+    ''', (user_id, track_id, event_id))
+
+
+def _create_conveners_entry(conn, user_id, track_id=None, event_id=None):
+    conn.execute('''
+        INSERT INTO events.track_conveners
+        (user_id, track_id, event_id) VALUES
+        (%s,       %s,      %s)
+    ''', (user_id, track_id, event_id))
+
+
+def _downgrade_permissions():
+    conn = op.get_bind()
+    query_track_permissions = 'SELECT user_id, track_id, permissions FROM events.track_principals'
+    for user_id, track_id, permissions in conn.execute(query_track_permissions):
+        for permission in permissions:
+            if permission == 'review':
+                _create_abstract_reviewers_entry(conn, user_id, track_id=track_id)
+            if permission == 'convene':
```

Comment:
`elif`? This way we will save an additional comparison

Reply:
but we can have both for one entry

Reply:
I'm gonna change those checks :)